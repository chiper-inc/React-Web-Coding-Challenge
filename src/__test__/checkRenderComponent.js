import { mount } from 'enzyme'
import { create } from 'react-test-renderer'

export const checkRenderComponent = (expect, component) => {
  const renderedComp = mount(component)
  expect(renderedComp.length).toEqual(1)
}

export const checkSnapShot = (expect, component) => {
  const renderComp = create(component)
  expect(renderComp.toJSON()).toMatchSnapshot()
}
