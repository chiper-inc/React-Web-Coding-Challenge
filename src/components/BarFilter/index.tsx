import { useState } from 'react';
import './style.scss';

const BarFilter = () => {
  return (
    <div className="bar-filter">
      <div className="content-search w-full md:w-3/12">
        <input
          className="input-search"
          placeholder="Search case descriptions"
        />
      </div>
      <div className="content-range-date"></div>
    </div>
  );
};

export default BarFilter;
