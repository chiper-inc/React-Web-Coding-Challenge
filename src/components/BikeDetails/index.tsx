import { useNavigate } from "react-router-dom";
import { useFormatDate } from "../../hooks/useFormatDate";
import Bike from "../../assets/images/bike.svg";
import { Wrapper, ContainerImg, ContainerInfo, Img, Row } from "./styles";

interface BikeDetailsProps {
  info: {
    id: number;
    thumb: string;
    title: string;
    description: string;
    date_stolen: number;
    stolen_location: string;
    frame_colors: string[];
    frame_material_slug: string;
    frame_model: string;
    frame_size: string;
    manufacturer_name: string;
    serial: string;
    type_of_cycle: string;
    year: number;
  }
}

export const BikeDetails = ({ info: {
  id,
  thumb,
  title,
  description,
  date_stolen,
  stolen_location,
  frame_colors,
  frame_material_slug,
  frame_model,
  frame_size,
  manufacturer_name,
  serial,
  type_of_cycle,
  year
}}: BikeDetailsProps) => {

  const navigate = useNavigate();
  const [date] = useFormatDate(date_stolen);

  const handleNavigate = () => navigate(`/case/${id}`);

  return (
    <Wrapper onClick={handleNavigate} >
      <ContainerImg>
        <Img src={thumb || Bike} alt={title} />
      </ContainerImg>
      <ContainerInfo>
        <span><b>Description of incident</b></span>
        <span>{description || 'No description.'}</span><br/>
        <Row>
          <span><b>Date Stolen:</b> {date || 'Unknown.'}</span>
          <span><b>Stolen Location:</b> {stolen_location || 'Unknown.'}</span>
        </Row>
        <Row>
          <span><b>Color(s):</b> {frame_colors.join(', ') || 'Unknown.'}</span>
          <span><b>Material:</b> {frame_material_slug || 'Unknown.'}</span>
        </Row>
        <Row>
          <span><b>Manufacturer:</b> {manufacturer_name || 'Unknown.'}</span>
          <span><b>Model:</b> {frame_model || 'Unknown.'}</span>
        </Row>
        <Row>
          <span><b>Serial:</b> {serial || 'Unknown.'}</span>
          <span><b>Size:</b> {frame_size?.toUpperCase() || 'Unknown.'}</span>
        </Row>
        <Row>
          <span><b>Type:</b> {type_of_cycle || 'Unknown.'}</span>
          <span><b>Year:</b> {year || 'Unknown.'}</span>
        </Row>
      </ContainerInfo>
    </Wrapper>
  );
};
