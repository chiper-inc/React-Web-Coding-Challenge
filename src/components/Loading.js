import React from "react";
import { Typography, CircularProgress } from "@material-ui/core";
import { LoadingContainer } from "../styles/LoadingStyles";

export const Loading = () => {
  return (
    <LoadingContainer>
      <CircularProgress />
      <Typography variant="h3" component="h3" color="primary">
        Loading...
      </Typography>
    </LoadingContainer>
  );
};
