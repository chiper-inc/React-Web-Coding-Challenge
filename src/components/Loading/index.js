import React from 'react'
import { OrbitalSpinnerStyled } from './styles'

export const OrbitalSpinner = () => <OrbitalSpinnerStyled color='#343a40' />
