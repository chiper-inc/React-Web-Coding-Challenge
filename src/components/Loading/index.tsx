import { Wrapper } from "./styles";
import Puff from "../../assets/images/puff.svg";

export const Loading = () => {
  return (
    <Wrapper>
      <img src={Puff} alt="loading"/>
    </Wrapper>
  );
};
