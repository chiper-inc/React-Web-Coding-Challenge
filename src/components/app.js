import React from 'react';
import IncidentsList from './incidents_list/incidents_list';
import styled from '../theme';

const Header = styled.header`
  background: #1d3f72;
  color: #fff;
  padding: 20px;
  margin: 0;
  text-align: center;
  border-bottom: 3px solid #71c2cc;

  h1, h2 {
    padding: 0;
    margin: 0;
    display: inline-block;
    vertical-align: middle;
  }

  h1 {
    background: url('logo.png') no-repeat;
    background-size: cover;
    width: 267px;
    height: 100px;
    overflow: hidden;
    text-indent: -999px;
    margin-right: 40px;
  }
`;

function App() {
  return (
    <div className="app">
      <Header>
        <h1>Berlin Polizei</h1>
        <h2>Stolen bykes</h2>
      </Header>
      <IncidentsList></IncidentsList>
    </div>
  );
}

export default App;
