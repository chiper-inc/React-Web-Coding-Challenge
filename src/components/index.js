import BarFilter from './BarFilter';
import Header from './Header';
import BikeItem from './BikeItem';
import Pagination from './Pagination';
import Loading from './Loading';
import ErrorData from './Error';

export { Header, BarFilter, BikeItem, Pagination, Loading, ErrorData };
