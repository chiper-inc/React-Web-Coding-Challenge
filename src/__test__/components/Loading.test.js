/* eslint-disable no-undef */
import React from 'react'
import { OrbitalSpinner } from '../../components/Loading'
import { checkRenderComponent, checkSnapShot } from '../checkRenderComponent'

describe('<OrbitalSpinner />', () => {
  test('Render of <OrbitalSpinner />', () => {
    checkRenderComponent(expect, <OrbitalSpinner />)
  })
})

describe('<OrbitalSpinner/> Snapshot', () => {
  test('Check UI of OrbitalSpinner', () => {
    checkSnapShot(expect, <OrbitalSpinner />)
  })
})
