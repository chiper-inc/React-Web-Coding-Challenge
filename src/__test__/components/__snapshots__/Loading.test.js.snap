// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrbitalSpinner/> Snapshot Check UI of OrbitalSpinner 1`] = `
<div
  className="styles__Container-sc-11w77it-0 czyLMx"
>
  <div
    className="lds-spinner styles__OrbitalSpinnerStyled-sc-11w77it-1 hraCVk"
    style={Object {}}
  >
    <div>
      <div
        className="div-after"
        style={
          Object {
            "background": "#343a40",
          }
        }
      />
    </div>
    <div>
      <div
        className="div-after"
        style={
          Object {
            "background": "#343a40",
          }
        }
      />
    </div>
    <div>
      <div
        className="div-after"
        style={
          Object {
            "background": "#343a40",
          }
        }
      />
    </div>
    <div>
      <div
        className="div-after"
        style={
          Object {
            "background": "#343a40",
          }
        }
      />
    </div>
    <div>
      <div
        className="div-after"
        style={
          Object {
            "background": "#343a40",
          }
        }
      />
    </div>
    <div>
      <div
        className="div-after"
        style={
          Object {
            "background": "#343a40",
          }
        }
      />
    </div>
    <div>
      <div
        className="div-after"
        style={
          Object {
            "background": "#343a40",
          }
        }
      />
    </div>
    <div>
      <div
        className="div-after"
        style={
          Object {
            "background": "#343a40",
          }
        }
      />
    </div>
    <div>
      <div
        className="div-after"
        style={
          Object {
            "background": "#343a40",
          }
        }
      />
    </div>
    <div>
      <div
        className="div-after"
        style={
          Object {
            "background": "#343a40",
          }
        }
      />
    </div>
    <div>
      <div
        className="div-after"
        style={
          Object {
            "background": "#343a40",
          }
        }
      />
    </div>
    <div>
      <div
        className="div-after"
        style={
          Object {
            "background": "#343a40",
          }
        }
      />
    </div>
  </div>
</div>
`;
